### ISSUE TITLE
Trying to set circular base type causes stack overflow.

### ISSUE THREAD

**Issue by pmeijer** [webgme/webgme#483]
```
Uncaught RangeError: Maximum call stack size exceeded 

isValidNode @ coretype.js:42core.getBase @ coretype.js:60core.getChildrenRelids @ coretype.js:196core.getChildrenRelids @ coretype.js:196core.getChildrenRelids @ coretype.js:196core.getChildrenRelids @ coretype.js:196core.getChildrenRelids
....
```

**Comment by pmeijer** [comment]
![circularbase](https://cloud.githubusercontent.com/assets/6518904/9231039/8034842c-40eb-11e5-8b02-fc4c0a3bcd18.PNG)

**Comment by pmeijer** [comment]
Closed by 47bf1e68b36586ba4321abe5fb4d1966c13bf73f
